### ISSUE TITLE
magento 2.3.5 not working on category page

### ISSUE THREAD

**Issue by rakeshdonga** [php-cuong/magento2-restrict-products#1]
i have used your extension and this extension working fine on product page but not working on category page 
can you check your end please and any change on that? please update here

**Comment by akgola** [comment]
@php-cuong , I have also used your extension , this is working fine on product page but not compatible for product listing page.Could you please check and update the code.

**Comment by eduardo-g-silva** [comment]
same here looks like a problem with the cache or something like that

**Comment by Luzmary** [comment]
yes, the problem is with cache from the catalog list maybe try with cacheable="false" on block class Magento\Catalog\Block\Product\ListProduct, and it will work fine
